app/services/db/models/public-cloud-request.ts

```diff
@@ -118,7 +118,9 @@ async function decorate<T extends PublicCloudRequestSimple | PublicCloudRequestD
     }
   }
 
-  const canCancel = doc.decisionStatus === DecisionStatus.PENDING && session.user.email === doc.createdBy?.email;
+  const canCancel =
+    doc.decisionStatus === DecisionStatus.PENDING &&
+    (session.user.email === doc.createdBy?.email || (getDaysSince(doc.createdAt) > 30 && session.isPublicAdmin));
```

Comment:
please create a new global permission instead of using role to validate things..